src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -69,6 +69,76 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
             }
         }
 
+        internal void CollectLocalsFromDeconstruction(
```

Comment:
I thought so too, but two things made things awkward:
1. The `ExpressionVariableFinder` creates variables of kind `RegularVariable` and the `ForeachLoopBinder` creates `ForEachIterationVariable`. 
2. The `ExpressionVariableFinder` is currently a visitor (it traverses down), but for it to work in a foreach loop, it would have to look up its parents to determine that it is in a foreach loop (and therefore should create a different kind of variable).

One option that is left would be to pass a lambda for a `collect` method that would handle all variables that are found.